src/cluster.c

```diff
@@ -562,6 +562,11 @@ int clusterLoadConfig(char *filename) {
     if (clusterGetMaxEpoch() > server.cluster->currentEpoch) {
         server.cluster->currentEpoch = clusterGetMaxEpoch();
     }
+
+    if (nodeIsSlave(myself) && myself->slaveof) {
```

Comment:
myself->slaveof implies nodeIsSlave(), if a node has a master but isn't marked as a slave that is an inconsistent state.
```suggestion
    if (myself->slaveof) {
```

Reply:
but we don't forbid a master has a master when load nodes.conf, and even more a node maybe flagged both master and slave. Do you think we should add some assertion in `clusterLoadConfig`? It's a breaking change but I think it's harmless, since it only break the invalid nodes.conf.

Reply:
I would assert that those two are consistent